modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/util/AssetVocabularyUtil.java

```diff
@@ -129,12 +133,26 @@ private static void _addAssetVocabulary(
 			Collection<Long> classNameIdsCollection) {
 
 		AssetVocabularySettingsHelper assetVocabularySettingsHelper =
-			new AssetVocabularySettingsHelper();
+			new AssetVocabularySettingsHelper(assetVocabulary.getSettings());
+
+		Set<Long> registeredClassNameIds = SetUtil.fromArray(
+			assetVocabularySettingsHelper.getRegisteredClassNameIds());
+
+		Set<Long> filteredClassNameIds = new LinkedHashSet<>();
+
+		for (long classNameId : classNameIdsCollection) {
+			if (!registeredClassNameIds.contains(classNameId)) {
+				filteredClassNameIds.add(classNameId);
+			}
+		}
```

Comment:
Why not simply do `classNameIdsCollection.removeAll(registeredClassNameIds)`? You can make a copy of the collection if it cannot be mutated in place.

Reply:
done